### ISSUE TITLE
<Epic> Fix color contrast issues to aline with ACT afw4f7: Text has minimum contrast

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#1195]
This is related to #1151, but has other [ACT test cases](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/afw4f7/proposed/) failed

1.  Engine context issue:  the context doesn't trigger the elements in a shadow dom:  ACT pass example 9
2.  color contrast calculation: when an element's background is a gradient, image, or shadow applied, the background cannot be calculated: ACT pass examples 2 & 3 & 4, and failure examples 2 & 3  & 7
3.  Color contrast check is only for human-readable text?  pass example 7
4.  aria-hidden && aria-disabled: inapplicable examples 3 & 7 & 9 & 11

Here are the test initial results for ACT test cases: 
pass_1:  passed
pass_2:   ignored, hasBackground = false;
pass_3:   ignored, hasBackground = false;
pass_4:   failed:  with ratio = 4.43, rather than 6.1 (text shadow)
pass_5:   passed
pass_6:   passed
pass_7:   failed with ratio = 3.66
pass_8:   passed
pass_9:   ignored, the engine context doesn’t include nodes in shadow
pass_10: passed
pass_11: passed.

fail_1:  passed
fail_2:   ignored, hasBackground = false;
fail_3:   ignored, hasBackground = false;
fail_4:   passed
fail_5:   passed
fail_6:   passed
fail_7:   failed with exception during ratio calculation
fail_8:   passed
fail_9:   passed
fail_10: passed

inapplicable_1:  passed
inapplicable_2:   passed
inapplicable_3:   failed with a potential (aria-hidden)
inapplicable_4:   passed
inapplicable_5:   passed
inapplicable_6:   passed
inapplicable_7:   ?  (aria-dsabled)
inapplicable_8:   passed
inapplicable_9:   ?  (aria-dsabled)
inapplicable_10: passed
inapplicable_11: ?  (aria-dsabled)

This is opened as an epic, and individual issues will be opened in this epic
